### PR TITLE
config consistency: add test for DD_HTTP_CLIENT_TAG_QUERY_STRING

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

- Adds shared tests for DD_HTTP_CLIENT_TAG_QUERY_STRING
- Skips shared for the following libraries: 
  - [java]( https://github.com/DataDog/dd-trace-java/blob/bb8b0d25a6fecd799ac15df28ff3f4d77e050de2/internal-api/src/main/java/datadog/trace/api/Config.java#L1291C13-L1291C41): sets DD_HTTP_CLIENT_TAG_QUERY_STRING to `False` by default.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_context/_scenarios/__init__.py [hunk 1]
```diff
@@ -470,10 +470,18 @@ def all_endtoend_scenarios(test_object):
...
+        "TRACING_CONFIG_EMPTY", weblog_env={}, doc="",
+    )
+
     tracing_config_nondefault = EndToEndScenario(
         "TRACING_CONFIG_NONDEFAULT", weblog_env={"DD_TRACE_HTTP_SERVER_ERROR_STATUSES": "200-201,202"}, doc="",
     )
 
+    tracing_config_nondefault_client_tag_query = EndToEndScenario(
+        "TRACING_CONFIG_NONDEFAULT_CLIENT_TAG_QUERY", weblog_env={"DD_HTTP_CLIENT_TAG_QUERY_STRING": "false"}, doc="",
+    )
```
Comment (mtoffl01) [inline utils/_context/_scenarios/__init__.py:483]: @cbeauchesne , we're running into an issue where DD_HTTP_CLIENT_TAG_QUERY_STRING and DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP interact when configured so we need to test them separately, meaning we need yet another, separate scenario. I know that's typically a problem. Do you have any suggestions?

Comment (cbeauchesne) [reply utils/_context/_scenarios/__init__.py:483]: Is it supposed to interact that way ? 

If yes, let's continue with `tracing_config_nondefault_3` 😓 (see #3007)

Comment (cbeauchesne) [reply utils/_context/_scenarios/__init__.py:483]: yes, sorry, I wasn't clear enough.

**Comment by cbeauchesne** [comment]
@mabdinur `./format.sh` is your friend

File: tests/test_config_consistency.py [hunk 2]
```diff
@@ -71,3 +71,38 @@ def test_status_code_202(self):
         assert spans[0]["type"] == "web"
         assert spans[0]["meta"]["http.status_code"] == "202"
         assert spans[0]["error"] == 1
+
+
+@scenarios.tracing_config_empty
+@features.tracing_configuration_consistency
+class Test_Config_ClientTagQueryString_Empty:
+    """ Verify behavior when set to empty string """
```
Comment (mtoffl01) [inline tests/test_config_consistency.py:79]:
```suggestion
    """ Verify behavior when DD_HTTP_CLIENT_TAG_QUERY_STRING set to empty string """
```

File: tests/test_config_consistency.py [hunk 3]
```diff
@@ -71,3 +71,38 @@ def test_status_code_202(self):
...
+    def test_query_string_redaction_unset(self):
+        client_request_span = _get_client_span(self.r, "/")
+        assert client_request_span
+        assert client_request_span["meta"].get("http.url") == "http://weblog:7777?hi=monkey"
+
+
+@scenarios.tracing_config_nondefault_3
+@features.tracing_configuration_consistency
+class Test_Config_ClientTagQueryString_Configured:
+    def setup_query_string_redaction(self):
```
Comment (mtoffl01) [inline tests/test_config_consistency.py:93]:
```suggestion
 """ Verify behavior when DD_HTTP_CLIENT_TAG_QUERY_STRING set to false """
    def setup_query_string_redaction(self):
```

**Review by mtoffl01 — CHANGES_REQUESTED** [review]
The scenarios are typically a list of env vars so it's good to specify the env var and value you're testing -- I added suggestions to make this clearer in your function defs.

Also, many of the manifests contain the `Test_Config_ClientTagQueryString_Configured` entry in the wrong place (some are under `parametric`, but they should not be), and all manifests are missing `Test_Config_ClientTagQueryString_Empty`

**Comment by cbeauchesne** [comment]
Rebase the branch to fix almost all errors ...

**Review by mtoffl01 — CHANGES_REQUESTED** [review]
It will be more seamless for the implementors to run these tests if the test entries are already in all the manfiests -- Can you add the test entries to all manifests, even if they're `missing_feature`? No need to test with all libs, we are treating `missing_feature` as a baseline.

Some PRs have just added "missing_feature" to all libs; meanwhile I tested my tests with just Java, which passed, so I:

- Added test entries with `missing_feature` to all manifests except java
- Added test entries to java with v1.39.0 (the latest version of dd-trace-java and the version I tested with, so I know it passes)

File: manifests/java.yml [hunk 4]
```diff
@@ -1231,6 +1231,7 @@ tests/:
   test_config_consistency.py:
     Test_Config_HttpServerErrorStatuses_Default: missing_feature
     Test_Config_HttpServerErrorStatuses_FeatureFlagCustom: missing_feature
+    Test_Config_ClientTagQueryString_Empty: incorrect default value (by default removes query string from http.url)
```
Comment (mabdinur) [inline manifests/java.yml:1234]:
```suggestion
    Test_Config_ClientTagQueryString_Empty: missing_feature  (incorrect default value)
```

File: manifests/java.yml [hunk 5]
```diff
@@ -1231,6 +1231,7 @@ tests/:
   test_config_consistency.py:
     Test_Config_HttpServerErrorStatuses_Default: missing_feature
     Test_Config_HttpServerErrorStatuses_FeatureFlagCustom: missing_feature
+    Test_Config_ClientTagQueryString_Empty: missing_feature (incorrect default value)
```
Comment (mabdinur) [inline manifests/java.yml:1234]:
```suggestion
    Test_Config_ClientTagQueryString_Empty: missing_feature (incorrect default value)
    Test_Config_HttpServerErrorStatuses_Default: missing_feature
    Test_Config_HttpServerErrorStatuses_FeatureFlagCustom: missing_feature
```

File: tests/test_config_consistency.py [hunk 6]
```diff
@@ -71,3 +71,40 @@ def test_status_code_202(self):
...
+        client_request_span = _get_client_span(self.r, "/")
+        assert client_request_span
+        assert client_request_span["meta"].get("http.url") == "http://weblog:7777/"
+
+
+def _get_client_span(r, endpoint):
+    for _, _, span in interfaces.library.get_spans(r, full_trace=True):
+        # Avoids retrieving the client span by the operation name, this value varies between languages
+        # Using span kind and resource is a better bet
+        if span["meta"].get("span.kind") == "client" and span["resource"] == f"GET {endpoint}":
```
Comment (mabdinur) [inline tests/test_config_consistency.py:108]:
```suggestion
        if span["resource"] == f"GET {endpoint}":
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
CI is failing : https://github.com/DataDog/system-tests/actions/runs/10888983033/job/30215264994?pr=3026#step:22:59

**Review by cbeauchesne — APPROVED** [review]

**Review by mtoffl01 — APPROVED** [review]

**Pull request merged**